src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs

```diff
@@ -144,11 +144,33 @@ protected virtual void InnerInvoke(IProgressTracker progressTracker, Cancellatio
             var extensionManager = Workspace.Services.GetService<IExtensionManager>();
             extensionManager.PerformAction(Provider, () =>
             {
+#if COCOA
+                InvokeWorkerAsync(getFromDocument, progressTracker, cancellationToken).Forget();
```

Comment:
So I don't get this code on the Cocoa side of things, since the .Forget() here the majority of this code may be running outside the PerformAction handler. But this is the current behavior.

Reply:
I'm following up to figure out if this hack is even still necessary, or whether VS for Mac now has a threaded wait dialog to use.

Reply:
This is also broken because the other stuff doing the caret position restoring and such isn't going to run correctly either.

Reply:
@DavidKarlas do you know if this is still necessary? How are we showing the message box now? I remember you added an out-of-proc UIThreadMonitorDaemon that would watch VSMac UI thread for hangs.

Reply:
I think thst Cocoa special code can be removed, since we now have ThreadedWaitDialog

Reply:
Alright I'll remove it then.

Reply:
And I'll do the removal in a separate commit, so if we have to put this back we'll be able to easily do so.